paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/complete_flow/CompleteFlowActivity.kt

```diff
@@ -0,0 +1,107 @@
+package com.stripe.android.paymentsheet.example.samples.ui.complete_flow
+
+import android.graphics.Color
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.material.MaterialTheme
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.snackbar.Snackbar
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.paymentsheet.PaymentSheet
+import com.stripe.android.paymentsheet.example.samples.ui.BuyButton
+import com.stripe.android.paymentsheet.example.samples.ui.Receipt
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+internal class CompleteFlowActivity : AppCompatActivity() {
+
+    private val snackbar by lazy {
+        Snackbar.make(findViewById(android.R.id.content), "", Snackbar.LENGTH_SHORT)
+            .setBackgroundTint(Color.BLACK)
+            .setTextColor(Color.WHITE)
+    }
+
+    private val viewModel by viewModels<CompleteFlowViewModel>()
+
+    private lateinit var paymentSheet: PaymentSheet
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        paymentSheet = PaymentSheet(
+            activity = this,
+            callback = viewModel::handlePaymentSheetResult,
+        )
+
+        setContent {
+            MaterialTheme {
+                val uiState by viewModel.state.collectAsState()
+
+                uiState.status?.let {
+                    LaunchedEffect(it) {
+                        snackbar.setText(it).show()
+                        viewModel.statusDisplayed()
+                    }
+                }
+
+                Receipt(
+                    isLoading = uiState.isProcessing,
+                    cartState = uiState.cartState,
+                    isEditable = false,
+                ) {
+                    BuyButton(
+                        buyButtonEnabled = !uiState.isProcessing,
+                        onClick = ::handleBuyButtonClick,
+                    )
+                }
+            }
+        }
+    }
+
+    private fun handleBuyButtonClick() {
+        lifecycleScope.launch {
+            val result = viewModel.prepareCheckout()
```

Comment:
Seems like this shouldn't be happening in the activity. We're waiting on a network request here.

Ideally this would happen as part of a viewModelScope, and stored as part of viewModelState.

Reply:
Yep, I guess you’re right. I wanted to keep things as simple as possible, but an approach more similar to the `CustomFlowActivity` seems better.